components/Activity/ActivityTransactions.tsx

```diff
@@ -114,24 +118,157 @@ export default function ActivityTransactions({ tab = ActivityTab.ALL }: Activity
     refetchBankTransfers,
   ]);
 
+  let fetchedEvents = useMemo(
+    () => activityData?.pages.flatMap(page => page?.docs || []) || [],
+    [activityData],
+  );
+
+  // Sync external transactions to database if they don't exist
   useEffect(() => {
-    if (!transactions || !user?.userId) return;
-    storeEvents(
-      user.userId,
-      transactions.map(transaction => ({
-        ...transaction,
-        clientTxId: `${transaction.type}-${transaction.hash}`,
-        amount: transaction.amount.toString(),
-        timestamp: parseInt(transaction.timestamp).toString(),
-      })),
+    if (!user?.userId || !transactions || transactions.length === 0) return;
+
+    transactions.forEach(async tx => {
```

Comment:
@LiorAgnin forEach will not await the async function, if you expect all transaction syncs to complete before proceeding, use Promise.all or map